src/AsyncWriter.js

```diff
@@ -390,22 +217,12 @@ AsyncWriter.prototype = {
         }
 
         lastArray.push(callback);
+        return this;
     },
 
     emit: function(type, arg) {
         var events = this._events;
-        switch(arguments.length) {
-            case 1:
-                events.emit(type);
-                break;
-            case 2:
-                events.emit(type, arg);
-                break;
-            default:
-                events.emit.apply(events, arguments);
-                break;
-        }
-
+        events.emit.apply(events, arguments);
```

Comment:
@patrick-steele-idem Looking through this again, I realized I removed this switch statement.  The functionality is the same with less code, but was this done for performance reasons?